src/main/java/net/logstash/logback/composite/AbstractFormattedTimestampJsonProvider.java

```diff
@@ -177,34 +174,78 @@ protected String getFormattedTimestamp(Event event) {
      * Updates the {@link #timestampWriter} value based on the current pattern and timeZone.
      */
     private void updateTimestampWriter() {
+        timestampWriter = createTimestampWriter();
+    }
+    
+    private TimestampWriter createTimestampWriter() {
         if (UNIX_TIMESTAMP_AS_NUMBER.equals(pattern)) {
-            timestampWriter = new NumberTimestampWriter();
-        } else if (UNIX_TIMESTAMP_AS_STRING.equals(pattern)) {
-            timestampWriter = new StringTimestampWriter();
-        } else if (pattern.startsWith("[") && pattern.endsWith("]")) {
-            String constant = pattern.substring("[".length(), pattern.length() - "]".length());
-            try {
-                Field field = DateTimeFormatter.class.getField(constant);
-                if (Modifier.isStatic(field.getModifiers())
-                        && Modifier.isFinal(field.getModifiers())
-                        && field.getType().equals(DateTimeFormatter.class)) {
-                    try {
-                        DateTimeFormatter formatter = (DateTimeFormatter) field.get(null);
-                        timestampWriter = new PatternTimestampWriter(formatter.withZone(timeZone.toZoneId()));
-                    } catch (IllegalAccessException e) {
-                        throw new IllegalArgumentException(String.format("Unable to get value of constant named %s in %s", constant, DateTimeFormatter.class), e);
-                    }
-                } else {
-                    throw new IllegalArgumentException(String.format("Field named %s in %s is not a constant %s", constant, DateTimeFormatter.class, DateTimeFormatter.class));
-                }
-            } catch (NoSuchFieldException e) {
-                throw new IllegalArgumentException(String.format("No constant named %s found in %s", constant, DateTimeFormatter.class), e);
+            return new NumberTimestampWriter();
+        }
+        
+        if (UNIX_TIMESTAMP_AS_STRING.equals(pattern)) {
+            return StringFormatterWriter.with(tstamp -> Long.toString(tstamp));
```

Comment:
would the `Long::toString` method reference work here instead of a lambda?

Reply:
Unfortunately not, the compiler will hesitate between `Long#toString()` and `Long#toString(long)` - so I needed to be explicit.